### DISCUSSION TITLE
Slow and clunky editors when using radix-ui

### DISCUSSION THREAD

**Discussion by maxzaleski** [radix-ui/primitives#810] [Help] [answered]
Anyone else experiencing a slow and clunky editor when using `radix-ui` types?

**Comment by jjenzz** [comment, marked as answer]
Yes. It's due to polymorphism. We've improved performance slightly recently but those changes haven't been released yet and there is still a lot of room for improvement. We just need to think of how to achieve it.

We've been thinking about providing a utility for enabling polymorphism instead of making everything polymorphic by default, because a lot of the time people don't need the polymorphism.

**Reply by maxzaleski** [reply]
Thanks. Will keep an eye out for the update!
